flink-table/flink-table-common/src/main/java/org/apache/flink/table/functions/BuiltInFunctionDefinitions.java

```diff
@@ -283,6 +283,26 @@
                     .outputTypeStrategy(nullable(explicit(DataTypes.BOOLEAN())))
                     .build();
 
+    public static final BuiltInFunctionDefinition GREATEST =
+            BuiltInFunctionDefinition.newBuilder()
+                    .name("greatest")
+                    .kind(SCALAR)
+                    .inputTypeStrategy(
+                            comparable(ConstantArgumentCount.from(1), StructuredComparision.FULL))
+                    .outputTypeStrategy(nullable(TypeStrategies.COMMON))
```

Comment:
Should we try to make the output type strategy more narrow by only making it nullable if any of the provided input arguments is nullable?

Reply:
IIUC, ` nullable(TypeStrategy initialStrategy)` has done for this? 
```
  /**
     * A type strategy that can be used to make a result type nullable if any of the input arguments
     * is nullable. Otherwise the type will be not null.
     */
    public static TypeStrategy nullable(TypeStrategy initialStrategy) {
        return nullable(ConstantArgumentCount.any(), initialStrategy);
    }
```

Reply:
Oh, you're right. :-)